packages/firestore/src/lite-api/aggregate.ts

```diff
@@ -15,40 +15,95 @@
  * limitations under the License.
  */
 
-import { DocumentData, Query, queryEqual } from './reference';
+import { CompositeFilterOpEnum } from '../protos/firestore_proto_api';
+import { invokeRunAggregationQueryRpc } from '../remote/datastore';
+import { hardAssert } from '../util/assert';
+import { cast } from '../util/input_validation';
+import { getDatastore } from './components';
+import { Firestore } from './database';
+import { Query, queryEqual } from './reference';
+import { LiteUserDataWriter } from './reference_impl';
 
-export class AggregateQuery<T = DocumentData> {
+/**
+ * A `AggregateQuery` computes some aggregation statistics from the result set of
+ * a base `Query`.
+ */
+export class AggregateQuery {
   readonly type = 'AggregateQuery';
-  readonly query: Query<T>;
+  /**
+   * The query on which you called `countQuery` in order to get this `AggregateQuery`.
+   * Query type is set to unknown to avoid error caused by query type converter.
+   * might change it back to T after testing if the error do exist or not
+   */
+  readonly query: Query<unknown>;
 
   /** @hideconstructor */
-  constructor(query: Query<T>) {
+  constructor(query: Query<unknown>) {
     this.query = query;
   }
 }
 
+/**
+ * A `AggregateQuerySnapshot` contains results of a `AggregateQuery`.
+ */
 export class AggregateQuerySnapshot {
   readonly type = 'AggregateQuerySnapshot';
   readonly query: AggregateQuery;
 
   /** @hideconstructor */
-  constructor(query: AggregateQuery, readonly _count: number) {
+  constructor(query: AggregateQuery, private readonly _count: number) {
     this.query = query;
   }
 
+  /**
+   * @return The result of a document count aggregation. Returns null if no count aggregation is
+   * available in the result.
+   */
   getCount(): number | null {
     return this._count;
   }
 }
 
-export function countQuery(query: Query): AggregateQuery {
+/**
+ * Creates an `AggregateQuery` counting the number of documents matching this query.
+ *
+ * @return An `AggregateQuery` object that can be used to count the number of documents in
+ * the result set of this query.
+ */
+export function countQuery(query: Query<unknown>): AggregateQuery {
   return new AggregateQuery(query);
 }
 
 export function getAggregateFromServerDirect(
-  query: AggregateQuery
+  aggregateQuery: AggregateQuery
```

Comment:
Please rename `aggregateQuery` back to `query`, to be consistent with the API proposal